src/main/java/org/jabref/gui/entryeditor/SourceTab.java

```diff
@@ -64,15 +63,17 @@
     private final BibDatabaseMode mode;
     private final UndoManager undoManager;
     private final ObjectProperty<ValidationMessage> sourceIsValid = new SimpleObjectProperty<>();
-    @SuppressWarnings("unchecked") private final ObservableRuleBasedValidator sourceValidator = new ObservableRuleBasedValidator(sourceIsValid);
+    private final ObservableRuleBasedValidator sourceValidator = new ObservableRuleBasedValidator();
     private final ImportFormatPreferences importFormatPreferences;
     private final FileUpdateMonitor fileMonitor;
     private final DialogService dialogService;
     private final StateManager stateManager;
     private Optional<Pattern> searchHighlightPattern = Optional.empty();
-    private CodeArea codeArea;
+    private final CodeArea codeArea;
     private KeyBindingRepository keyBindingRepository;
 
+    private BibEntry boundEntry;
```

Comment:
The parent `EntryEditorTab` already defines `currentEntry`, which seems to be what you want.

Reply:
changed the Name to previousEntry. Part of the fix for the recreated CodeArea. JabRef needs to keep the previous entry to store it, because the focus event is fired after the new entry is loaded.